pandas/core/internals.py

```diff
@@ -3233,6 +3236,16 @@ def comp(s):
             return _possibly_compare(values, getattr(s, 'asm8', s),
                                      operator.eq)
 
+        def _cast(block, scalar):
```

Comment:
this might be a more general routine (IOW maybe defin as a utility function in internals rather than a nested function here), or maybe as a block method?

``cast_scalar``?